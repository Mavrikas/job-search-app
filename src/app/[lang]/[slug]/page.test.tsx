import React from 'react';
import { render } from '@testing-library/react';
import JobDetails from './page';
import '@testing-library/jest-dom';

// Mocks
jest.mock('next/link', () => {
  const MockLink = ({ children, href }: any) => <a href={href}>{children}</a>;
  MockLink.displayName = 'MockLink';
  return MockLink;
});
jest.mock('primereact/chip', () => ({
  Chip: ({ label }: any) => <span>{label}</span>,
}));
jest.mock('@/app/components/ApplyButton/ApplyButton', () => ({
  ApplyButton: () => <button>Apply</button>,
}));
jest.mock('./styles.module.css', () => ({}));
jest.mock('@/store/uitls', () => ({
  formatDate: (date: string) => `formatted-${date}`,
}));
const mockDictionary = {
  somethingWrong: 'Something went wrong',
  jobNotFound: 'Job not found',
  returnHome: 'Return Home',
};
jest.mock('@/hooks/useDictionary', () => ({
  useDictionary: () => mockDictionary,
}));

const mockJob = {
  id: '1',
  slug: 'test-job',
  title: 'Test Job Title',
  company: 'Test Company',
  description: 'Test job description',
  location: 'Test Location',
  category: 'Test Category',
  tags: ['tag1', 'tag2'],
  postedAt: '2024-06-01',
};

const mockJobsResponse = {
  results: [mockJob],
};

global.fetch = jest.fn();

describe('JobDetails', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders job details when job is found', async () => {
    (fetch as jest.Mock).mockResolvedValueOnce({
      json: async () => mockJobsResponse,
    });
    const params = Promise.resolve({ slug: 'test-job', lang: 'en' });
    const { asFragment } = render(await JobDetails({ params }));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders not found message when job is not found', async () => {
    (fetch as jest.Mock).mockResolvedValueOnce({
      json: async () => ({ results: [] }),
    });
    const params = Promise.resolve({ slug: 'not-exist', lang: 'en' });
    const { asFragment } = render(await JobDetails({ params }));
    expect(asFragment()).toMatchSnapshot();
  });
});
