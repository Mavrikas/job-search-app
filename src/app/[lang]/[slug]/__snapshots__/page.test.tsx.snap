// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDetails renders job details when job is found 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <main>
      <div
        class="border-solid border-3 rounded-lg p-5 bg-[#fafafa] max-w-[800px] min-w-[400px] m-1"
      >
        <h1
          class="text-2xl font-bold"
        >
          Test Job Title
        </h1>
        <h3
          class="text-lg font-medium"
        >
          Test Company
        </h3>
        <div
          class="flex flex-row justify-start items-center gap-2 mt-2"
        >
          <span>
            Test Location
          </span>
          <span>
            formatted-2024-06-01
          </span>
        </div>
        <span>
          Test Category
        </span>
        <p
          class="text-lg mt-3 max-w-[600px]"
        >
          Test job description
        </p>
        <div
          class="flex flex-row justify-start items-center gap-2 mt-10"
        >
          <span>
            tag1
          </span>
          <span>
            tag2
          </span>
        </div>
        <div
          class="flex flex-row items-center justify-end"
        >
          <button>
            Apply
          </button>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`JobDetails renders not found message when job is not found 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <main>
      <div
        class="border-solid border-3 rounded-lg p-5 bg-[#fafafa] max-w-[800px] min-w-[400px] m-1"
      >
        <h1
          class="text-2xl font-bold"
        >
          Something went wrong.
        </h1>
        <p
          class="text-lg mt-3 max-w-[600px] mb-4"
        >
          Job not found.
        </p>
        <a
          href="/en"
        >
          Return to Home
        </a>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
